[MNOE-689] API V2 - SubTenant remove mass assignment

x4d3 opened the pull request:
- Introduce filterParams parameter of organization and user lists directives
- Remove unused staff-local-list
- Use parameter to retrieve the scoped organizations for a staff or a sub_tenant
- Use the dedicated update_clients, update_account_managers methods instead of relying on array of ids

alexnoox commented on src/app/components/mnoe-organizations-list/mnoe-organizations-list.coffee:
```diff
@@ -1,10 +1,10 @@
-@App.directive('mnoeOrganizationsList', ($filter, $log, MnoeOrganizations, MnoeAdminConfig, MnoeCurrentUser) ->
-  restrict: 'E'
+@App.directive('mnoeOrganizationsList', ($filter, $log, MnoeOrganizations, MnoeAdminConfig, MnoeCurrentUser, MnoeObservables, OBS_KEYS) ->
+  restrict: 'E',
```
You don't need a comma here

alexnoox commented on src/app/views/staff/staff.controller.coffee:
```diff
@@ -19,10 +20,8 @@
-        MnoeSubTenants.get(vm.staff.mnoe_sub_tenant_id).then((r) -> vm.staff.subTenants = r.data.sub_tenants)
   )
+  # Temporary solution, does not scale if there is more than 50 subtenants
   MnoeSubTenants.list(null, null, null).then((response) -> vm.subTenants = response.data)
```
Why can't this be paginated?

x4d3 replied:
Because it's too fill a Combo Box.

alexnoox replied:
Then the solution would be to reuse `angucomplete-alt` lib. Already used in this project.

alexnoox commented on src/app/views/staff/update-staff-clients-modal/update-staff-clients.controller.coffee:
```diff
@@ -50,30 +48,56 @@
+        vm.changes.remove.push(organization.id)
+      else
+        vm.changes.add.splice(index, 1)
+    console.log(JSON.stringify(vm.changes))
```
Remove `console.log()`

alexnoox commented on src/app/views/staff/update-staff-clients-modal/update-staff-clients.controller.coffee:
```diff
@@ -50,30 +48,56 @@
   vm.onCancel = () ->
     $uibModalInstance.dismiss('cancel')
 
+  vm.isDisable = () ->
```
Should be `vm.isDisabled`

alexnoox commented on src/app/views/sub-tenant/update-account-managers-modal/update-account-managers.controller.coffee:
```diff
@@ -78,4 +76,31 @@
+  vm.checkBoxChanged = (user) ->
+    if user.belong_to_sub_tenant
+      index = _.indexOf(vm.changes.remove, user.id)
+      if index == -1
```
Use `_.contains()` for more code clarity. Also, it stops looping at the first occurrence.

x4d3 replied:
Yeah, but I need the index to remove the element.

alexnoox replied:
👍

x4d3 commented:
@alexnoox I've addressed your comments. And added the angucomplete library. I had to take some liberty as I did not manage to link the initial-value to the model.

alexnoox commented on src/app/views/staff/staff.controller.coffee:
```diff
@@ -14,16 +15,30 @@
+  vm.subTenantSelected = (subTenant) ->
+    if subTenant
+      vm.staff.sub_tenant_id = subTenant.originalObject.id
+      console.log(vm.staff.sub_tenant_id)
```
Log to remove

alexnoox commented on src/app/components/mnoe-organizations-list/mnoe-organizations-list.coffee:
```diff
@@ -80,12 +80,23 @@
+    this.$onDestroy = ->
+      MnoeObservables.unsubscribe(OBS_KEYS.organizationChanged, onOrganizationChanged)
+
+
```
Should be only one empty line

x4d3 commented:
This is waiting on https://github.com/maestrano/mno-enterprise/pull/532

ouranos commented:
Reopened in #245
